paddle/fluid/operators/lerp_op.h

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include "paddle/fluid/framework/op_registry.h"
+#include "paddle/fluid/framework/operator.h"
+
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif
+
+namespace paddle {
+namespace operators {
+
+static framework::DDim GetNewDims(const framework::DDim& in_dims, int rank) {
```

Comment:
Some questions:

1. It looks like GetSubDims is a better name. In software engineer, new** usually means create a new object.
2. Should we put this to `ddim.h/ddim.cc`
3. Does the `slice_dim` in `ddim.h` meet your requirement? If yes, please re-use the code.

Reply:
This function is used to extend original dims to larger rank. Get eigen tensor via `framework::EigenTensor<T, D>::From` will check rank of dims, so we need get correctly rank dims before calling it.

Using `ExtendDims2Rank` now may be a better name.